utilities/write_batch_with_index/write_batch_with_index.cc

```diff
@@ -79,6 +79,7 @@ class BaseDeltaIterator : public Iterator {
   void Next() override {
     if (!Valid()) {
       status_ = Status::NotSupported("Next() on invalid iterator");
+      return;
```

Comment:
I think an this `return` was intended here, because without it the `Advance()` below would fail an assert.